packages/shared/friends/sagas.ts

```diff
@@ -2487,3 +2523,78 @@ function buildFriendRequestErrorResponse(error: FriendshipErrorCode) {
 function buildFriendRequestReply<T>(reply: NonNullable<T>) {
   return { reply, error: undefined }
 }
+
+/**
+ * Check whether the user has reached the max number allowed of sent requests to a given user. If not, update the friends state.
+ * @param userId - the user to check the number of requests for.
+ * @returns true if the user has reached the max number of sent requests, false otherwise
+ */
+function reachedMaxNumberOfRequests(userId: string) {
+  // Get number friend requests sent in the current session to the given user
+  const sentRequests = getNumberOfFriendRequests(store.getState())
+  const number = sentRequests.get(userId) ?? 0
+
+  // Get the maximum number of requests allowed
+  const maxNumber = DEFAULT_MAX_NUMBER_OF_REQUESTS
+
+  // Check if the current number of requests is less than the maximum allowed
+  if (number < maxNumber) {
+    // Update the number of sent requests and return false
+    sentRequests.set(userId, number + 1)
+    return false
+  }
+  return true
+}
+
+/**
+ * Check whether there is a remaining cooldown time to send a friend request to a given user. If not, update the friends state.
+ * @param userId - the user to check the cooldown time for.
+ * @returns true if there is a remaining cooldown time and it hasn't expired yet, false otherwise
+ */
+function hasRemainingCooldown(userId: string) {
+  const currentTime = Date.now()
+
+  // Get the remaining cooldown time for the given user
+  const coolDownTimer = getCoolDownOfFriendRequests(store.getState())
+  const remainingCooldownTime = coolDownTimer.get(userId)
+
+  const cooldownWindow = COOLDOWN_TIME
+
+  // If there is a remaining cooldown time and it hasn't expired yet, return false
+  if (remainingCooldownTime && currentTime < remainingCooldownTime) {
+    return true
+  } else {
+    // Update the cooldown timer and return false
+    coolDownTimer.set(userId, currentTime + cooldownWindow)
+    return false
+  }
+}
+
+/**
+ * Filters and processes (update their friendship status as rejected) friend requests from blocked users.
+ * @param fromFriendRequests - an array of from friend requests.
+ */
+function handleBlockedUsers(fromFriendRequests: FriendRequest[]) {
+  // Get the ids of users who have been blocked
+  const blockedIds = fromFriendRequests.filter((fromFriendRequest) => isBlocked(fromFriendRequest.userId))
+
+  // For each blocked user, update their friendship status as rejected
+  blockedIds.map(
+    async (fromFriendRequest) =>
+      await UpdateFriendshipAsPromise(
+        FriendshipAction.REJECTED,
+        fromFriendRequest.userId.toLowerCase(),
+        false,
+        fromFriendRequest.message
+      )
+  )
+}
```

Comment:
Let's return a `Promise#all` of the above created promises so that the client of this function can actually await on it if need be!